storage/allocator_test.go

```diff
@@ -293,3 +289,438 @@ func TestExistingReplica(t *testing.T) {
+			if err := g.AddInfo(key, testStores[j].StoreDescriptor, 0); err != nil {
+				panic(err)
+			}
+		}
+		wg.Wait()
+
+		// Next loop through test stores and maybe rebalance. Only one rebalance
+		// can occur to a node per cycle.
+		active := map[proto.StoreID]struct{}{}
+		for j := 0; j < len(testStores); j++ {
+			ts := &testStores[j]
+			if alloc.ShouldRebalance(&testStores[j].StoreDescriptor) {
+				target, err := alloc.Rebalance(proto.Attributes{}, []proto.Replica{{NodeID: ts.Node.NodeID, StoreID: ts.StoreID}})
+				if err == nil {
+					if _, ok := active[target.StoreID]; !ok {
+						active[target.StoreID] = struct{}{}
+						testStores[j].Rebalance(&testStores[int(target.StoreID)], alloc.randGen.Int63n(1<<20))
+					}
+				}
+			}
+		}
+
+		// Output store capacities as hexidecimal 2-character values.
+		if i%(generations/20) == 0 {
+			var maxBytes int64
+			for j := 0; j < len(testStores); j++ {
+				bytes := testStores[j].Capacity.Capacity - testStores[j].Capacity.Available
+				if bytes > maxBytes {
+					maxBytes = bytes
+				}
+			}
+			if maxBytes > 0 {
+				for j := 0; j < len(testStores); j++ {
+					endStr := " "
+					if j == len(testStores)-1 {
+						endStr = ""
+					}
+					bytes := testStores[j].Capacity.Capacity - testStores[j].Capacity.Available
+					fmt.Printf("%02x%s", (255*bytes)/maxBytes, endStr)
+				}
+				fmt.Printf("\n")
+			}
+		}
+	}
+
+	var totBytes int64
+	var totRanges int32
+	for i := 0; i < len(testStores); i++ {
+		totBytes += testStores[i].Capacity.Capacity - testStores[i].Capacity.Available
+		totRanges += testStores[i].Capacity.RangeCount
+	}
+	fmt.Printf("Total bytes=%d, ranges=%d\n", totBytes, totRanges)
+
+	// Output:
+	// ff 00 00 00 00 00 00 bc 00 00 00 00 00 00 00 00 00 00 00 00
+	// e9 bb 34 df 1e 79 68 80 70 6b a3 ff c1 e1 60 a1 97 d5 49 a4
+	// e4 dc ca eb 60 8b 96 a9 7f ff 55 db a3 ed e2 98 98 8f d9 b5
+	// df b7 e4 d1 78 bb ba ae a5 d7 7e df c2 ff f1 b5 b6 b1 e2 b7
+	// c9 be ff dd 8f d1 b7 c0 a0 cd 8b e4 c1 f0 db bd b2 be e5 c4
+	// c8 bf f7 ae 8e d2 d4 dd a7 c2 94 db c2 ff e6 b5 b8 dd e7 ce
+	// d2 cb f6 bd ab d1 ce dc aa c1 a4 c7 c3 ff e2 b7 ca e8 e8 cd
+	// d4 ca e4 c9 bd dc c9 d2 ac cd af bf db ff e8 bc e1 e1 e3 cc
+	// dc c9 e4 ce c8 d8 c7 cf b9 d1 bc c4 e5 ff dc b6 dd e2 e1 c7
+	// ea c6 da dc da cc d9 c5 c3 db c1 c3 ef ff e8 c0 d8 e2 e7 c6
+	// f6 cc e2 eb ed d8 e0 c8 d2 ed ca c7 f5 f8 ea cc e6 f1 ff c8
+	// ff cc da e8 f2 d7 d7 d3 d4 e9 ce cb ec f5 eb cd dc e7 fc bd
+	// ff d3 e4 f2 ee da dc d9 d7 f0 da c9 f2 f3 f0 d6 de e6 fd cd
+	// ff ce e5 f0 f7 df d8 e4 d3 f0 d8 c7 f0 eb f1 db db e1 fe c9
+	// f9 c8 e6 f6 f7 e3 d8 e4 cd f2 d7 c8 ee f2 f4 df d4 e4 ff cd
+	// fd d1 e9 f6 f7 e2 d7 e0 cf ef d8 ce f6 f2 f9 e0 dc ec ff d0
+	// fc d2 e8 f5 fd e7 da e7 d6 f6 e2 d3 ff f8 f9 e7 e0 f2 f9 d0
+	// ee d8 df f2 f8 e4 dc e0 d5 f2 dd d2 ff f2 f3 e7 de e5 f9 d3
+	// ee e0 e4 f0 ff ec db e4 d6 f4 dc d7 fd ef ee eb df e9 fd cf
+	// f1 e4 e7 ec fa ee e0 e9 d5 fa dd dd ff f0 f2 ef e3 f0 fc d1
```

Comment:
I was thinking 3 digit percentages from 0-100. Or if you know you'll never reach 100% you could do 0-99.